Makefile

```diff
@@ -659,6 +659,9 @@ cpplint:
 pylint:
 	python3 -m pylint --rcfile=$(ROOTDIR)/ci/other/pylintrc --ignore-patterns=".*\.so$$,.*\.dll$$,.*\.dylib$$" python/mxnet tools/caffe_converter/*.py
 
+sample_lib:
+	$(CXX) -shared -fPIC example/lib_api/mylib.cc -o libsample_lib.so -I include/mxnet
```

Comment:
`libsample_lib.so` cannot be cleaned via `make clean`. Could you please move it to the lib folder?

Reply:
I dont think we want to move it to the lib folder, since this library is not used for linking or typical LD_LIBRARY_PATH stuff. What if we just add a rm line to the clean target to delete this? We can do it as part of the custom op PR #15921

Reply:
Thank you for the explanation, @samskalicky . I'm just wondering if we should put a `.so` file in the root folder and I feel difficult to understand the usage of the `.so` from its name.

Reply:
@TaoLv this has been fixed in the custom Op PR from @rondogency by adding entries to the clean target:
https://github.com/apache/incubator-mxnet/pull/15921/files#diff-b67911656ef5d18c4ae36cb6741b7965R763

Reply:
Thank you for fixing that.